edge-agent/src/Microsoft.Azure.Devices.Edge.Agent.Edgelet/versioning/ModuleManagementHttpClientVersioned.cs

```diff
@@ -108,6 +109,17 @@ public virtual async Task<Stream> GetModuleLogs(string module, bool follow, Opti
                     async () =>
                     {
                         HttpResponseMessage httpResponseMessage = await httpClient.SendAsync(httpRequest, HttpCompletionOption.ResponseHeadersRead, cancellationToken);
+                        if (httpResponseMessage != null && !httpResponseMessage.IsSuccessStatusCode)
+                        {
+                            switch (httpResponseMessage.StatusCode)
+                            {
+                                case HttpStatusCode.BadRequest:
+                                    throw new ArgumentException($"Request Returned Status Code {httpResponseMessage.StatusCode} with Message {await httpResponseMessage.Content.ReadAsStringAsync()}");
+                                default:
+                                    throw new InvalidOperationException($"Request Returned Status Code {httpResponseMessage.StatusCode} with Message {await httpResponseMessage.Content.ReadAsStringAsync()}");
+                            }
```

Comment:
I'm not sure why, but GetModuleLogs was added directly to the base class (ModuleManagementHttpClientVersioned) instead of adding it to ModuleManagementHttpClient under the latest `iotedge/edge-agent/src/Microsoft.Azure.Devices.Edge.Agent.Edgelet/version_*` directory at the time.

What that means is that this function has to do its own error handling instead of using all the error handling built from the swagger interface. But I wonder if it can/should still throw `EdgeletCommunicationException` instead of `ArgumentException` and `InvalidOperationException`, something like [this](https://github.com/Azure/iotedge/blob/e15b86b2783d64b6136b15fcd09e3584269c3be1/edge-agent/src/Microsoft.Azure.Devices.Edge.Agent.Edgelet/version_2020_07_07/ModuleManagementHttpClient.cs#L227-L248)? Let me do a little more investigation...

Reply:
I think i should be able to use EdgeletCommunicationException. I will verify that.

Reply:
Doesn't EdgeletCommunicationException use whatever status code you passed into the constructor?

Reply:
I dont think it uses the status code for anything other than displaying the message. 

```
[13:22:09 VRB] Method 'GetModuleLogs' on 'e2e-7c801001001-211116-132023.507-001/$edgeAgent' returned: 500
{"message":"{\"message\":\"invalid parameter: since\"}"}
```

The Code that you referenced is in case Swagger throws an exception and that will continue to be the case if the API Version we use throws an exception ( eg - Edgelet doesnt return a response)

In our case, we do get the response back from edgelet.

Reply:
I think this method is not using the generated code because of how it handles the parameters to build the url, it could still be moved if PrepareRequest is used to handle special case. If not move, it should throw EdgeletCommunicationException which has a message and a status code

Reply:
The Problem with ```EdgeletCommunicationException```  is that it masks the problem. The Exception returns a ```500``` status code which is an internal server error. If the Format is wrong, we should throw a ```400``` right?

Reply:
@nimanch : Where do you see EdgeletCommunicationException return a 500?

Reply:
@varunpuranik 

I verified that in the same E2E Test and pasted the response in my comment above

https://github.com/Azure/iotedge/pull/5846#discussion_r750672617

Reply:
@varunpuranik @ancaantochi , sorry to bug you again :), but do you have any other comments on this?

Reply:
Had an offline conversation with @ancaantochi and agreed to throw the default exception as `EdgeletCommunicationException'. The Higher Layer [RequestManager](https://github.com/nimanch/iotedge/blob/fbeec72cf974b0b37f45de83fb6a395ba79b6029/edge-agent/src/Microsoft.Azure.Devices.Edge.Agent.Core/requests/RequestManager.cs#L66-L76) catches the `ArgumentException` and returns a `Bad Request` , For all other exceptions we return a `500`

Reply:
+1 on that - that is the right thing to do. Thanks!